### PR TITLE
[docs] update urls throughout documentation

### PR DESCRIPTION
**Pull request by jhamman**
We are in the process of moving our documentation url from https://xarray.pydata.org to https://docs.xarray.dev. This PR makes that change throughout the documentation. Additionally, I corrected some broken links and fixed some missing https urls in the process.

cc @andersy005

### CODE REVIEW

**Review by andersy005 — APPROVED** [review]
Thank you, @jhamman! This looks great...

File: doc/internals/how-to-add-new-backend.rst [hunk 1]
```diff
@@ -311,9 +311,7 @@ The BackendArray subclass shall implement the following method and attributes:
 - the ``shape`` attribute
 - the ``dtype`` attribute.
 
-
-Xarray supports different type of
-`indexing <http://xarray.pydata.org/en/stable/indexing.html>`__, that can be
+Xarray supports different type of `indexing <../user-guide/indexing.html>`__, that can be
```
Comment (keewis) [inline doc/internals/how-to-add-new-backend.rst:314]: wouldn't it be more reliable to use a rst `:doc:` reference?

File: doc/whats-new.rst [hunk 2]
```diff
@@ -4781,7 +4781,7 @@ scientists who work with actual x-rays are interested in using this project in
 their work. Thanks for your understanding and patience in this transition. You
 can now find our documentation and code repository at new URLs:
 
-- http://xarray.pydata.org
+- https://docs.xarray.dev
 - http://github.com/pydata/xarray/
```
Comment (andersy005) [inline doc/whats-new.rst:4785]:
```suggestion
- https://github.com/pydata/xarray/
```

File: xarray/backends/api.py [hunk 3]
```diff
@@ -834,8 +834,8 @@ def open_mfdataset(
     References
     ----------
 
-    .. [1] http://xarray.pydata.org/en/stable/dask.html
-    .. [2] http://xarray.pydata.org/en/stable/dask.html#chunking-and-performance
+    .. [1] http://docs.xarray.dev/en/stable/dask.html
+    .. [2] http://docs.xarray.dev/en/stable/dask.html#chunking-and-performance
```
Comment (andersy005) [inline xarray/backends/api.py:838]:
```suggestion
    .. [1] https://docs.xarray.dev/en/stable/dask.html
    .. [2] https://docs.xarray.dev/en/stable/dask.html#chunking-and-performance
```

File: xarray/backends/plugins.py [hunk 4]
```diff
@@ -126,23 +126,23 @@ def guess_engine(store_spec):
...
+                "http://docs.xarray.dev/en/stable/user-guide/io"
             )
     else:
         error_msg = (
             "found the following matches with the input file in xarray's IO "
             f"backends: {compatible_engines}. But their dependencies may not be installed, see:\n"
-            "http://xarray.pydata.org/en/stable/user-guide/io.html \n"
-            "http://xarray.pydata.org/en/stable/getting-started-guide/installing.html"
+            "http://docs.xarray.dev/en/stable/user-guide/io.html \n"
+            "http://docs.xarray.dev/en/stable/getting-started-guide/installing.html"
```
Comment (andersy005) [inline xarray/backends/plugins.py:145]:
```suggestion
            "https://docs.xarray.dev/en/stable/user-guide/io.html \n"
            "https://docs.xarray.dev/en/stable/getting-started-guide/installing.html"
```

**Pull request merged**